chef_master/source/resource_archive_file.rst

```diff
@@ -0,0 +1,234 @@
+=====================================================
+archive_file resource
+=====================================================
+`[edit on GitHub] <https://github.com/chef/chef-web-docs/blob/master/chef_master/source/resource_archive_file.rst>`__
+
+Use the **archive_file** resource to extract archive files to disk. This resource uses the libarchive library to extract multiple archive formats including tar, gzip, bzip, and zip formats.
+
+**New in Chef Client 15.0.**
+
+Syntax
+=====================================================
+The archive_file resource has the following syntax:
+
+.. code-block:: ruby
+
+  archive_file 'name' do
+    destination      String
+    group            String
+    mode             String, Integer # default value: "755"
+    options          Array, Symbol
+    overwrite        true, false, auto # default value: false
+    owner            String
+    path             String # default value: 'name' unless specified
+    action           Symbol # defaults to :extract if not specified
+  end
+
+where:
+
+* ``archive_file`` is the resource.
+* ``name`` is the name given to the resource block.
+* ``action`` identifies which steps the chef-client will take to bring the node into the desired state.
+* ``destination``, ``group``, ``mode``, ``options``, ``overwrite``, ``owner``, and ``path`` are the properties available to this resource.
+
+Actions
+=====================================================
+
+The archive_file resource has the following actions:
+
+``:extract``
+    Description here.
+
+``:nothing``
+   .. tag resources_common_actions_nothing
+
+   This resource block does not act unless notified by another resource to take action. Once notified, this resource block either runs immediately or is queued up to run at the end of the Chef Client run.
+
+   .. end_tag
+
+Properties
+=====================================================
+
+The archive_file resource has the following properties:
+
+``destination``
+   **Ruby Type:** String | ``REQUIRED``
```

Comment:
Is the location of this ``REQUIRED`` a new format? I was under the impression this type of information would be included in the description instead.

Reply:
It's auto generated from resource data. We missed most of them previously when we relied on the author to put it in the description wording.